File: src/main/java/fr/spoutnik87/musicbot_rest/model/Bot.java
package fr.spoutnik87.musicbot_rest.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonView;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;

@NoArgsConstructor
@RequiredArgsConstructor
@Data
@Entity
@Table(name = "Bot")
public class Bot extends AuditModel implements Serializable {

  @JsonView(Views.Public.class)
  @NonNull
  @JsonProperty("id")
  @Column(nullable = false, unique = true)
  private String uuid;

  @JsonView(Views.Public.class)
  @NonNull
  @Column(nullable = false)
  private String name;

  @JsonView(Views.Public.class)
  @NonNull
  @Column(nullable = false)
  private String token;

  @JsonView(Views.Public.class)
  @OneToOne(cascade = CascadeType.ALL)
  @JoinColumn(name = "server_id")
  @JsonManagedReference
  private Server server;
}

File: src/main/kotlin/fr/spoutnik87/musicbot_rest/model/Bot.kt
package fr.spoutnik87.musicbot_rest.model

import com.fasterxml.jackson.annotation.JsonManagedReference
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonView
import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "Bot")
data class Bot(
        @JsonView(Views.Companion.Public::class)
        @JsonProperty("id")
        @Column(nullable = false, unique = true)
        var uuid: String,
        @JsonView(Views.Companion.Public::class)
        @Column(nullable = false)
        var name: String,
        @JsonView(Views.Companion.Public::class)
        @Column(nullable = false)
        var token: String
) : AuditModel(), Serializable {
    @JsonView(Views.Companion.Public::class)
    @OneToOne(cascade = [CascadeType.ALL])
    @JoinColumn(name = "server_id")
    @JsonManagedReference
    lateinit var server: Server

    constructor(uuid: String, name: String, token: String, server: Server) : this(uuid, name, token) {
        this.server = server
    }
}